ui/admin/app/components/form/credential-library/vault-ssh-cert/index.js

```diff
@@ -1,4 +1,69 @@
-/* eslint-disable ember/no-empty-glimmer-component-classes */
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action, set } from '@ember/object';
+import {
+  options,
+  TYPES_CREDENTIAL_LIBRARY,
+} from 'api/models/credential-library';
 
-export default class FormCredentialLibraryVaultSshCertComponent extends Component {}
+export default class FormCredentialLibraryVaultSshCertComponent extends Component {
+  // =properties
+
+  @tracked
+  showKeyBits = false;
+  @tracked
+  newCriticalOptionKey;
+  @tracked
+  newCriticalOptionValue;
+
+  /**
+   * @type {object}
+   */
+  keyTypes = options.key_type;
+
+  /**
+   * @type {Array.<string>}
+   */
+  types = TYPES_CREDENTIAL_LIBRARY;
+
+  /**
+   * Action to determine whether the key bits field should be displayed on the form.
+   * @param value {string}
+   */
+  @action
+  shouldDisplayKeyBits({ target: { value } }) {
+    if (value === 'ed25519') {
+      this.showKeyBits = false;
+      this.args.model.key_type = null;
+    } else {
+      this.showKeyBits = true;
+      this.args.mode.key_type = value;
+    }
+  }
```

Comment:
There are a couple things going on here that could be separated:

- Set `key_type` on the model independently of any display logic.
- Convert `showKeyBits` to a getter that is based on the value of `key_type`.

Then `shouldDisplayKeyBits` can be dropped.

Reply:
Good call, updated it